lib/src/entrypoint.dart

```diff
@@ -1017,4 +1024,55 @@ See https://dart.dev/go/sdk-constraint
   /// results.
   bool get _summaryOnlyEnvironment =>
       (Platform.environment['PUB_SUMMARY_ONLY'] ?? '0') != '0';
+
+  /// Checks if the Flutter SDK is available and whether this is a Flutter app
+  /// package.
+  bool shouldRunFlutterPackageHook() {
+    if (!FlutterSdk().isAvailable) {
+      return false;
+    }
+
+    // TODO: sky_engine check
+    final hasFlutterDependency = root.dependencies.values.any((package) {
```

Comment:
We should make sure this also works for transitive dependencies. (Maybe a test-case?)

Reply:
Gotcha! looking into it